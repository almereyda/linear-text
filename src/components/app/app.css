.App-header {
  background: var(--color-light-grey);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}
