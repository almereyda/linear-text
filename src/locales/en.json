{
  "button-add-record__label": {
    "translation": "Add",
    "extractedComments": []
  },
  "button-add-record__title": {
    "translation": "Create new record [alt-shift-enter].",
    "extractedComments": []
  },
  "button-help__label": {
    "translation": "Help",
    "extractedComments": []
  },
  "button-help__title": {
    "translation": "Open documentation [alt-shift-f1].",
    "extractedComments": []
  },
  "button-load-file__label": {
    "translation": "Open…",
    "extractedComments": []
  },
  "button-load-file__title": {
    "translation": "Load an existing file [alt-shift-o].",
    "extractedComments": []
  },
  "button-new-file__label": {
    "translation": "New",
    "extractedComments": []
  },
  "button-new-file__title": {
    "translation": "Create an empty file [alt-shift-n].",
    "extractedComments": []
  },
  "button-redo__label": {
    "translation": "Redo",
    "extractedComments": []
  },
  "button-redo__title": {
    "translation": "Reverse an undo [alt-shift-y].",
    "extractedComments": []
  },
  "button-remove-record__label": {
    "translation": "Discard",
    "extractedComments": []
  },
  "button-remove-record__title": {
    "translation": "Remove selected record(s) [alt-shift-delete].",
    "extractedComments": []
  },
  "button-save-file-as__label": {
    "translation": "Save As…",
    "extractedComments": []
  },
  "button-save-file-as__title": {
    "translation": "Open save dialog.",
    "extractedComments": []
  },
  "button-save-file__label": {
    "translation": "Save",
    "extractedComments": []
  },
  "button-save-file__title": {
    "translation": "Download file [alt-shift-s].",
    "extractedComments": []
  },
  "button-toggle-toolbar__label": {
    "translation": "Menu",
    "extractedComments": []
  },
  "button-toggle-toolbar__title": {
    "translation": "Toggle overflow menu.",
    "extractedComments": []
  },
  "button-undo__label": {
    "translation": "Undo",
    "extractedComments": []
  },
  "button-undo__title": {
    "translation": "Restore the previous state [alt-shift-z].",
    "extractedComments": []
  },
  "title-with-{version}": {
    "translation": "Perfect Memory v{version}",
    "extractedComments": []
  }
}