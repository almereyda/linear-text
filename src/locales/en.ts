/*eslint-disable*/export const messages={"app-title-with-saved":"Linear Text","app-title-with-saved-{filename}":[["filename"]," – Linear Text"],"app-title-with-unsaved":"• (Unsaved) – Linear Text","app-title-with-unsaved-{filename}":["• ",["filename"]," (Unsaved) – Linear Text"],"app-title-with-{version}":["Linear Text v",["version"]],"button-add-group__access-key":"|","button-add-group__label":"Group","button-add-group__title":"Add group [alt–|].","button-add-line__access-key":"+","button-add-line__label":"Add","button-add-line__title":"Add new line [alt–+].","button-dismiss-dialog__label":"Dismiss","button-dismiss-dialog__title":"Dismiss dialog.","button-help__access-key":"?","button-help__label":"Help","button-help__title":"Open documentation [alt–?].","button-load-file__access-key":"o","button-load-file__label":"Open…","button-load-file__title":"Load an existing file [alt–o].","button-new-file__access-key":"n","button-new-file__label":"New","button-new-file__title":"Create an empty file [alt–n].","button-redo__access-key":"y","button-redo__label":"Redo","button-redo__title":"Reverse an undo [alt–y].","button-remove-line__access-key":"-","button-remove-line__label":"Discard","button-remove-line__title":"Remove selected line(s) [alt–-].","button-save-file-as__label":"Save As…","button-save-file-as__title":"Open save dialog.","button-save-file__access-key":"s","button-save-file__label":"Save","button-save-file__title":"Download file [alt–s].","button-undo__access-key":"z","button-undo__label":"Undo","button-undo__title":"Restore the previous state [alt–z].","dropdown-file-type__description":"Plain text files"};