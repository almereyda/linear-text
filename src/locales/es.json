{
  "app-title-saved-{filename}": {
    "translation": "",
    "extractedComments": []
  },
  "app-title-unsaved-{filename}": {
    "translation": "",
    "extractedComments": []
  },
  "app-title-with-{version}": {
    "translation": "",
    "extractedComments": []
  },
  "button-add-line__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-add-line__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-help__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-help__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-load-file__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-load-file__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-new-file__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-new-file__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-redo__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-redo__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-remove-line__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-remove-line__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-save-file-as__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-save-file-as__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-save-file__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-save-file__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-toggle-menu__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-toggle-toolbar__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-toggle-toolbar__title": {
    "translation": "",
    "extractedComments": []
  },
  "button-undo__label": {
    "translation": "",
    "extractedComments": []
  },
  "button-undo__title": {
    "translation": "",
    "extractedComments": []
  },
  "dropdown-file-type__description": {
    "translation": "",
    "extractedComments": []
  }
}