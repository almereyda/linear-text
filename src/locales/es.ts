/*eslint-disable*/export const messages={"app-title-with-saved":"app-title-with-saved","app-title-with-saved-{filename}":["app-title-with-saved-",["filename"]],"app-title-with-unsaved":"app-title-with-unsaved","app-title-with-unsaved-{filename}":["app-title-with-unsaved-",["filename"]],"app-title-with-{version}":["app-title-with-",["version"]],"button-add-group__access-key":"button-add-group__access-key","button-add-group__label":"button-add-group__label","button-add-group__title":"button-add-group__title","button-add-line__access-key":"button-add-line__access-key","button-add-line__label":"button-add-line__label","button-add-line__title":"button-add-line__title","button-dismiss-dialog__label":"button-dismiss-dialog__label","button-dismiss-dialog__title":"button-dismiss-dialog__title","button-grab-line__title":"button-grab-line__title","button-help__access-key":"button-help__access-key","button-help__label":"button-help__label","button-help__title":"button-help__title","button-load-file__access-key":"button-load-file__access-key","button-load-file__label":"button-load-file__label","button-load-file__title":"button-load-file__title","button-move-line__description":"button-move-line__description","button-new-file__access-key":"button-new-file__access-key","button-new-file__label":"button-new-file__label","button-new-file__title":"button-new-file__title","button-redo__access-key":"button-redo__access-key","button-redo__label":"button-redo__label","button-redo__title":"button-redo__title","button-remove-line__access-key":"button-remove-line__access-key","button-remove-line__label":"button-remove-line__label","button-remove-line__title":"button-remove-line__title","button-save-file-as__label":"button-save-file-as__label","button-save-file-as__title":"button-save-file-as__title","button-save-file__access-key":"button-save-file__access-key","button-save-file__label":"button-save-file__label","button-save-file__title":"button-save-file__title","button-undo__access-key":"button-undo__access-key","button-undo__label":"button-undo__label","button-undo__title":"button-undo__title","dropdown-file-type__description":"dropdown-file-type__description"};