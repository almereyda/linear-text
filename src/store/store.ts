import {Action, configureStore, ThunkAction} from '@reduxjs/toolkit'
import {tableSlice} from './table-slice/table-slice'
import undoable, {excludeAction, UndoableOptions} from 'redux-undo'

const undoConfig: Readonly<UndoableOptions> = Object.freeze({
  clearHistoryType: [
    tableSlice.actions.newFileAction.type,
    tableSlice.actions.saveFileAction.type
  ],
  filter: excludeAction([
    tableSlice.actions.addDividerAction.type,
    tableSlice.actions.addDraftAction.type,
    tableSlice.actions.focusLineAction.type
  ])
})

export const store = configureStore({
  reducer: {table: undoable(tableSlice.reducer, undoConfig)}
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
