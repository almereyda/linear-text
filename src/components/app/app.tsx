import {Group} from '../group/group'
import {MenuCard} from '../menu/menu'
import {selectRecords} from '../../store/records/records-slice'
import {useAppSelector} from '../../hooks/use-store'

import './app.css'

export type AppProps = Readonly<{'data-testid'?: string}>

export function App({'data-testid': testID}: AppProps) {
  const records = useAppSelector(selectRecords)
  return (
    <div className='app' data-testid={testID}>
      <header className='app__header'>
        <MenuCard />
      </header>
      <main className='app__body'>
        <Group lines={records.tab.records} />
      </main>
    </div>
  )
}
