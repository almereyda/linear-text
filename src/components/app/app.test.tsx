import {App} from './app'
import {en} from 'make-plural/plurals'
import {i18n} from '@lingui/core'
import {I18nProvider} from '@lingui/react'
import {Provider} from 'react-redux'
import {render, screen} from '@testing-library/react'
import {store} from '../../store/store'
import {messages as stringsEn} from '../../locales/en'

beforeEach(() => {
  i18n.loadLocaleData('en', {plurals: en})
  i18n.load({en: stringsEn})
  i18n.activate('en')
})

test('matches snapshot', () => {
  render(
    <Provider store={store}>
      <I18nProvider i18n={i18n}>
        <App data-testid='app' />
      </I18nProvider>
    </Provider>
  )
  expect(screen.getByTestId('app')).toMatchSnapshot()
})
