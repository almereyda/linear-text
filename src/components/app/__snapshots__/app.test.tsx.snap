// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="app"
  data-testid="app"
>
  <header
    class="app__header"
  >
    <div
      class="menu-card"
    >
      <div
        class="card"
      >
        <nav
          class="menu"
        >
          <div
            class="menu-list"
          >
            <ul
              class="unordered-list unordered-list--horizontal"
            >
              <li
                class="list-item"
              >
                <details
                  class="disclosure"
                >
                  <summary
                    class="disclosure__summary buttonish "
                    data-title="Toggle overflow menu."
                    role="button"
                  >
                    <img
                      alt=""
                      class="icon"
                      decoding="sync"
                      height="20"
                      loading="eager"
                      src="menu-icon-toggle-toolbar.svg"
                      width="20"
                    />
                  </summary>
                  <div
                    class="toolbar-card"
                  >
                    <div
                      class="bubble-card"
                    >
                      <div
                        class="card"
                      >
                        <ul
                          class="unordered-list unordered-list--grid"
                        >
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Restore the previous state [alt-shift-z]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-undo.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Undo
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Reverse an undo [alt-shift-y]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-redo.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Redo
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Download file [alt-shift-s]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-save-file.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Save
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Open save dialog."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-save-file-as.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Save As…
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Load an existing file [alt-shift-o]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-load-file.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Open…
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Create an empty file [alt-shift-n]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-new-file.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                New
                              </label>
                            </div>
                          </li>
                          <li
                            class="list-item"
                          >
                            <div
                              class="icon-button"
                            >
                              <button
                                accesskey=""
                                class="buttonish "
                                data-title="Open documentation [alt-shift-f1]."
                              >
                                <img
                                  alt=""
                                  class="icon"
                                  decoding="sync"
                                  height="20"
                                  loading="eager"
                                  src="toolbar-icon-help.svg"
                                  width="20"
                                />
                              </button>
                              <label
                                class="icon-buttonish__label"
                              >
                                Help
                              </label>
                            </div>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </details>
                <label
                  class="icon-buttonish__label"
                >
                  Menu
                </label>
              </li>
              <li
                class="list-item"
              >
                <div
                  class="icon-button"
                >
                  <button
                    accesskey=""
                    class="buttonish "
                    data-title="Remove selected record(s) [alt-shift-delete]."
                  >
                    <img
                      alt=""
                      class="icon"
                      decoding="sync"
                      height="20"
                      loading="eager"
                      src="menu-icon-remove-record.svg"
                      width="20"
                    />
                  </button>
                  <label
                    class="icon-buttonish__label"
                  >
                    Discard
                  </label>
                </div>
              </li>
              <li
                class="list-item"
              >
                <div
                  class="icon-button"
                >
                  <button
                    accesskey=""
                    class="buttonish "
                    data-title="Create new record [alt-shift-enter]."
                  >
                    <img
                      alt=""
                      class="icon"
                      decoding="sync"
                      height="20"
                      loading="eager"
                      src="menu-icon-add-record.svg"
                      width="20"
                    />
                  </button>
                  <label
                    class="icon-buttonish__label"
                  >
                    Add
                  </label>
                </div>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="app__body"
  />
</div>
`;
